src/bacnet/basic/object/netport.c

```diff
@@ -1956,51 +1879,62 @@ bool Network_Port_Remote_BBMD_IP_Address_Set(
 }
 
 /**
- * For a given object instance-number, gets the BBMD UDP Port number
- * Note: depends on Network_Type being set to PORT_TYPE_BIP for this object
+ * For a given object instance-number, sets the FD BBMD Address:
+ * either as IP address or a hostname.
+ * Note: depends on Network_Type being set for this object
  *
  * @param  object_instance - object-instance number of the object
+ * @param  bbmd_address - BACNET_HOST_N_PORT FD_BBMD_Address
  *
- * @return BBMD UDP Port number
+ * @return  true if BBMD Address was set
  */
-uint16_t Network_Port_Remote_BBMD_BIP_Port(uint32_t object_instance)
+bool Network_Port_Remote_BBMD_Address_Set(
+    uint32_t object_instance, const BACNET_HOST_N_PORT *const bbmd_address)
 {
-    uint16_t value = 0;
-    unsigned index = 0;
+    bool status = false;
+    unsigned index = Network_Port_Instance_To_Index(
+        object_instance); /* offset from instance lookup */
 
-    index = Network_Port_Instance_To_Index(object_instance);
     if (index < BACNET_NETWORK_PORTS_MAX) {
-        if (Object_List[index].Network_Type == PORT_TYPE_BIP) {
-            value = Object_List[index].Network.IPv4.BBMD_Port;
-        }
-    }
+        BACNET_HOST_N_PORT *dest_bbmd_address = NULL;
 
-    return value;
-}
+        if (Object_List[index].Network_Type == PORT_TYPE_BIP) {
+            dest_bbmd_address = &Object_List[index].Network.IPv4.BBMD_Address;
+        } else if (Object_List[index].Network_Type == PORT_TYPE_BIP6) {
+            dest_bbmd_address = &Object_List[index].Network.IPv6.BBMD_Address;
+        }
+
+        if (dest_bbmd_address) {
+            if ((dest_bbmd_address->host_name != bbmd_address->host_name) ||
```

Comment:
Can `host_n_port_same()` be used instead?

Reply:
done.